packages/slope/src/adapter.ts

```diff
@@ -0,0 +1,191 @@
+import {
+    BaseSignerWalletAdapter,
+    pollUntilReady,
+    WalletAccountError,
+    WalletNotConnectedError,
+    WalletNotFoundError,
+    WalletNotInstalledError,
+    WalletPublicKeyError,
+    WalletSignTransactionError,
+} from '@solana/wallet-adapter-base';
+import { PublicKey, Transaction } from '@solana/web3.js';
+import bs58 from 'bs58';
+
+interface SlopeWallet {
+    connect(): Promise<{
+        msg: string;
+        data: {
+            publicKey: string;
+        };
+    }>;
+    disconnect(): Promise<{
+        msg: string;
+    }>;
+    signTransaction(message: string): Promise<{
+        msg: string;
+        data: {
+            publicKey: string;
+            signature: string;
+        }
+    }>;
+    signAllTransactions(messages: string[]): Promise<{
+        msg: string;
+        data: {
+            publicKey: string;
+            signatures: string[];
+        }
+    }>;
+}
+
+interface SlopeWindow extends Window {
+    Slope?: any;
+}
+
+declare const window: SlopeWindow;
+
+export interface SlopeWalletAdapterConfig {
+    pollInterval?: number;
+    pollCount?: number;
+}
+
+export class SlopeWalletAdapter extends BaseSignerWalletAdapter {
+    private _connecting: boolean;
+    private _wallet: SlopeWallet | null;
+    private _publicKey: PublicKey | null;
+
+    constructor(config: SlopeWalletAdapterConfig = {}) {
+        super();
+        this._connecting = false;
+        this._wallet = null;
+        this._publicKey = null;
+
+        if (!this.ready) pollUntilReady(this, config.pollInterval || 1000, config.pollCount || 3);
+    }
+
+    get publicKey(): PublicKey | null {
+        return this._publicKey;
+    }
+
+    get ready(): boolean {
+        return !!window.Slope;
```

Comment:
```suggestion
        return typeof window !== 'undefined' && !!window.Slope;
```

I added some checks in https://github.com/solana-labs/wallet-adapter/pull/32 to handle server side rendering, where the window object isn't available.